src/Twine/Loop.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+module Twine.Loop (
+    loop
+  ) where
+
+import           Control.Monad.IO.Class (MonadIO, liftIO)
+
+import           P
+
+import           Twine.Data.Check
+import           Twine.Data.Duration
+import           Twine.Snooze
+
+
+-- | Loop with a delay until the continue check fails
+--
+loop :: MonadIO m => Duration -> Check -> m () -> m ()
+loop d c action = do
+  action
```

Comment:
Should we care if `action` throws an exception? or assume it's been wrapped up in some handler already